content/en/docs/concepts/services-networking/service.md

```diff
@@ -316,6 +316,32 @@ This field follows standard Kubernetes label syntax. Values should either be
 [IANA standard service names](https://www.iana.org/assignments/service-names) or
 domain prefixed names such as `mycompany.com/my-custom-protocol`.
 
+### Endpoint readiness
+
+In most proxy modes, if the endpoint Pod that kube-proxy redirects to does not
+respond, the connection fails. (This was _not_ the case with the original
+"userspace" mode, where kube-proxy would detect that the connection to the first
+Pod had failed and would automatically retry with a different backend Pod.)
+
+You can use Pod [readiness probes](/docs/concepts/workloads/pods/pod-lifecycle/#container-probes)
+to verify that backend Pods are working OK, so that kube-proxy only sees
+backends that test out as healthy. Doing this means you avoid having traffic
+sent via kube-proxy to a Pod that's known to have failed.
+
+### Endpoint affinity
+
+If you want to make sure that connections from a particular client
+are passed to the same Pod each time, you can select the session affinity based
+on the client's IP addresses by setting `service.spec.sessionAffinity` to "ClientIP"
+(the default is "None").
+You can also set the maximum session sticky time by setting
+`service.spec.sessionAffinityConfig.clientIP.timeoutSeconds` appropriately.
+(the default value is 10800, which works out to be 3 hours).
+
+{{< note >}}
+On Windows, setting the maximum session sticky time for Services is not supported.
+{{< /note >}}
```

Comment:
It'd be nice to see these changes to the Virtual IPs page go in as part of a separate PR. Ideally, update https://kubernetes.io/docs/reference/networking/virtual-ips/#session-affinity to point to this text.

Reply:
removed this from this PR